api-lib/profileImages.ts

```diff
@@ -0,0 +1,84 @@
+import { VercelRequest } from '@vercel/node';
+
+import {
+  composeHasuraActionRequestBodyWithSession,
+  HasuraUserSessionVariables,
+  uploadImageInput,
+} from '../src/lib/zod';
+
+import { gql } from './Gql';
+
+export const userAndImageData = (
+  req: VercelRequest
+): {
+  input: { image_data_base64: string };
+  hasuraProfileId: number;
+} => {
+  const {
+    input: { object: input },
+    session_variables: sessionVariables,
+  } = composeHasuraActionRequestBodyWithSession(
+    uploadImageInput,
+    HasuraUserSessionVariables
+  ).parse(req.body);
+  return { input, hasuraProfileId: sessionVariables.hasuraProfileId };
+};
+
+export const profileUpdateAvatarMutation = (hasuraProfileId: number) => {
+  return profileUpdateImageMutation(hasuraProfileId, (fileName: string) => {
+    return { avatar: fileName };
+  });
+};
+
+export const profileUpdateBackgroundMutation = (hasuraProfileId: number) => {
+  return profileUpdateImageMutation(hasuraProfileId, (fileName: string) => {
+    return { background: fileName };
+  });
+};
+
+const profileUpdateImageMutation = (
```

Comment:
It would be better to organize mutations / queries in a subfolder under api-lib, instead of spreading them out across multiple files in a flat way. Will be easier to maintain and track that way. e.g. having a single `imageHelpers.ts` file that contains all the non graphql stuff, then having `api-lib/mutations/profile.ts` that contains profile update mutations etc. 

Also, for naming functions should try to word them in a way that describes what they **do**. e.g. for this, a better name would be `createProfileUpdateImageMutation`